snapshotter/utils/generic_worker.py

```diff
@@ -67,11 +92,135 @@ def __init__(self, name, **kwargs):
         self._commit_payload_exchange = (
             f'{settings.rabbitmq.setup.commit_payload.exchange}:{settings.namespace}'
         )
+        self._event_detector_exchange = f'{settings.rabbitmq.setup.event_detector.exchange}:{settings.namespace}'
+        self._event_detector_routing_key_prefix = f'powerloom-event-detector:{settings.namespace}:{settings.instance_id}.'
         self._commit_payload_routing_key = (
             f'powerloom-backend-commit-payload:{settings.namespace}:{settings.instance_id}.Data'
         )
         self._initialized = False
 
+    def _signal_handler(self, signum, frame):
+        if signum in [SIGINT, SIGTERM, SIGQUIT]:
+            self._core_rmq_consumer.cancel()
+
+    @retry(
+        wait=wait_random_exponential(multiplier=1, max=10),
+        stop=stop_after_attempt(5),
+        retry=tenacity.retry_if_not_exception_type(httpx.HTTPStatusError),
+        after=web3_storage_retry_state_callback,
+    )
+    async def _upload_web3_storage(self, snapshot: bytes):
```

Comment:
Web3Storage uploads have always been optional, I think we should keep it that way i.e if Auth token is not set, skip web3storage uploading.